src/Tasks.UnitTests/FormatUrl_Tests.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable disable
+
+namespace Microsoft.Build.UnitTests
+{
+    sealed public class FormatUrl_Tests
+    {
+        private readonly ITestOutputHelper _out;
+
+        public FormatUrl_Tests(ITestOutputHelper testOutputHelper)
+        {
+            _out = testOutputHelper;
+        }
+
+        /// <summary>
+        /// The URL to format is null.
+        /// </summary>
+        [Fact]
+        public void NullTest()
+        {
+            var t = new FormatUrl();
+            t.BuildEngine = new MockEngine(_out);
+
+            t.InputUrl = null;
+            var expected = string.Empty;
+
+            Assert.True(t.Execute()); // "success"
+#if DEBUG
+            _out.WriteLine("InputUrl " + ((null == t.InputUrl) ? "is null." : $"= '{t.InputUrl}'."));
+            _out.WriteLine("expected " + ((null == expected) ? "is null." : $"= '{expected}'."));
+            _out.WriteLine("OutputUrl " + ((null == t.OutputUrl) ? "is null." : $"= '{t.OutputUrl}'."));
+#endif
+            Assert.Equal(expected, t.OutputUrl);
```

Comment:
We've shifted to preferring Shouldly in tests that are new or recently touched because it helps simplify this sort of thing by having clearer assert messages. Would you mind adopting this pattern?

```suggestion
            t.InputUrl = null;

            t.Execute().ShouldBeTrue();

            t.OutputUrl.ShouldBe(string.Empty);
```

Reply:
Changed from Xunit asserts to Shouldly asserts.